bin/fillTestData/Experiences.js

```diff
@@ -0,0 +1,342 @@
+/**
+ * Required dependencies
+ */
+const _ = require('lodash');
+const path = require('path');
+const mongooseService = require(path.resolve('./config/lib/mongoose'));
+const chalk = require('chalk');
+const yargs = require('yargs');
+const faker = require('faker');
+const mongoose = require('mongoose');
+const config = require(path.resolve('./config/config'));
+const util = require(path.resolve('./bin/fillTestData/util'));
+
+/**
+ * Configure the script usage using yargs to obtain parameters and enforce usage.
+ */
+const argv = yargs.usage(
+  '$0 <averageNumberPerProfile> <replyRate>',
+  'Seed database with experiences',
+  yargs => {
+    return yargs
+      .positional('averageNumberPerProfile', {
+        describe:
+          'Average Number of experiences a profile will have ' +
+          '(except for pre-populated profiles)',
+        type: 'number',
+      })
+      .positional('replyRate', {
+        describe:
+          'The probability that B will share experience with A if A shares one with B',
+        type: 'number',
+      })
+      .boolean('debug')
+      .describe('debug', 'Enable extra database output (default=false)')
+      .boolean('force')
+      .describe(
+        'force',
+        'Remove existing experiences if there are any and create new ones',
+      )
+      .example(
+        '$0 10 0.8',
+        'Adds on average 10 experiences (8 of them having replies) to each profile to the database ',
+      )
+      .example(
+        '$0 10 0.8 --force',
+        'Adds on average 10 experiences (8 of them having replies) to each profile to the database after removing all existing references from the db ',
+      )
+      .example(
+        '$0 10 0.8 --debug',
+        'Adds on average 10 experiences (8 of them having replies) to each profile to the database with debug database output',
+      )
+      .check(function (argv) {
+        if (argv.replyRate > 1 || argv.replyRate < 0) {
+          throw new Error('Reply rate must be between 0 and 1');
+        }
+        if (argv.averageNumberPerProfile < 0) {
+          throw new Error('Number of experiences must be positive');
+        }
+        return true;
+      })
+      .strict().yargs;
+  },
+).argv;
+
+const profileType = {
```

Comment:
I think I have over-engineered things here a little, and also to have these magic numbers is not nice and robust enough. But in the end, I thought for a 'dev' task it might be good enough for now. Let me know what you think.

Reply:
You could always add a comment above the object to elaborate more on numbers if you feel like it. :-)